addons/mrp/models/mrp_bom.py

```diff
@@ -432,6 +432,102 @@ def _set_outdated_bom_in_productions(self):
             if productions:
                 productions.is_outdated_bom = True
 
+    # -------------------------------------------------------------------------
+    # CATALOG
+    # -------------------------------------------------------------------------
+
+    def action_add_from_catalog(self, **kwargs):
+        res = super().action_add_from_catalog()
+        if res['context'].get('product_catalog_order_model') == self._name:
+            res['context']['scope'] = kwargs['scope']
+
+        return res
```

Comment:
Perhaps `child_model` is also not ideal because it is possible we might have same child model (e.g. `stock.move`) but still need to differentiate between two scopes (e.g. components and byproducts inside MO). Not sure what to name it then